TaskManager/TaskManager/WeeklyTaskService.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TaskManagerNamespace
+{
+    internal class WeeklyTaskService
+    {
+        private int _counter;
+        internal readonly WeeklyTask[] _tasks;
+
+        public WeeklyTaskService()
+        {
+            _tasks = new WeeklyTask[10];
+        }
+
+        public void HandleAddNewTask()
+        {
+            if (_counter >= 10)
+            {
+                Console.WriteLine("Out of memory. Delete unnecessary tasks, and try again");
+                return;
+            }
+
+            Console.WriteLine("Add new task in format: {},{},{}");
+            var inputData = Console.ReadLine();
+            var task = ParseNewTask(inputData);
+            AddNewTask(task);
+        }
+
+        public void HandleList()
+        {
+            for (int i = 0; i < _counter; i++)
+            {
+                PrintTask(_tasks[i], i);
+            }
+        }
+
+        public void HandleEdit()
+        {
+            Console.WriteLine("Input number to edit:");
+            var inputNumber = Console.ReadLine();
+            var taskNumber = int.Parse(inputNumber);
+
+            Console.WriteLine("Input new task data in format: {},{},{}:");
+            var inputTaskData = Console.ReadLine();
+            var task = ParseNewTask(inputTaskData);
+            _tasks[taskNumber - 1] = task;
```

Comment:
extract a well named method: 
```suggestion
            UpdateTask(taskNumber - 1, task)
```

Reply:
extract